File: Dante1/Module1.07/Zadanie7.07/main.c
#include <stdio.h>

//add, subtract, multiply, divide

float add(float a, float b)
{
    return a + b;
}

float subtract(float a, float b)
{
    return a - b;
}

float divide(float a, float b)
{
    if(b == 0)
    {
        return -1;
    }
    return a / b;
}

float multiply(float a, float b)
{
    return a * b;
}

int main()
{
    float num1, num2;

    printf("Enter num1: ");

    int check1 = scanf("%f",&num1);
    if(check1 != 1)
    {
        printf("Incorrect input");
        return 1;
    }

    printf("Enter num2: ");

    int check2 = scanf("%f",&num2);
    if(check2 != 1)
    {
        printf("Incorrect input");
        return 1;
    }

    printf("Suma: %.2f\n", add(num1, num2));
    printf("Roznica: %.2f\n", subtract(num1, num2));
    printf("Iloczyn: %.2f\n", multiply(num1, num2));
    float result = divide(num1, num2);
    if(result == -1 && (num2 != -1 && num1 != -1))
    {
        printf("Operation not permitted");
        return 0;
    }
    else
    {
        printf("Iloczyn: %.2f\n", result);
    }
    return 0;
}

File: Dante1/Module1.07/Zadanie7.07/main.rs
fn add(a: f32, b: f32) -> f32 {
    a + b
}

fn subtract(a: f32, b: f32) -> f32 {
    a - b
}

fn multiply(a: f32, b: f32) -> f32 {
    a * b
}

fn divide(a: f32, b: f32) -> f32 {
    if b == 0.0 {
        return -1.0;
    }
    a / b
}

fn main() {
    let mut input: String = String::new();

    println!("Enter: ");
    std::io::stdin().read_line(&mut input).expect("Error");
    let a: f32 = input.trim().parse().expect("Error");

    input.clear();

    println!("Enter: ");
    std::io::stdin().read_line(&mut input).expect("Error");
    let b: f32 = input.trim().parse().expect("Error");

    println!("Add: {}", add(a, b));
    println!("Subtract: {}", subtract(a, b));
    println!("Multiply: {}", multiply(a, b));

    let result: f32 = divide(a, b);
    if result == -1.0 {
        panic!("Operation not permitted");
    }
    else {
        println!("Divide: {}", result);
    }
}
